object/service.proto

```diff
@@ -595,6 +536,76 @@ message SearchResponse {
   neo.fs.v2.session.ResponseVerificationHeader verify_header = 3;
 }
 
+// Object SearchV2 request
+message SearchV2Request {
+  // Object Search request body
+  message Body {
+    // Container where the search is being performed.
+    neo.fs.v2.refs.ContainerID container_id = 1;
+
+    // Version of the Query Language used.
+    uint32 version = 2;
+
+    // List of search expressions. Limited to 8.
+    repeated SearchFilter filters = 3;
+
+    // Cursor to continue search. Can be omitted or empty for the new search.
+    string cursor = 4;
+
+    // Limits the number of responses to the specified number. Can't be more
+    // than 1000.
+    uint32 count = 5;
+
+    // List of attribute names to include into the reply. Limited to 4,
+    // these attributes also affect result ordering.
+    repeated string attributes = 6;
+  }
+  // Body of search object request message.
+  Body body = 1;
+
+  // Carries request meta information. Header data is used only to regulate
+  // message transport and does not affect request execution.
+  neo.fs.v2.session.RequestMetaHeader meta_header = 2;
+
+  // Carries request verification information. This header is used to
+  // authenticate the nodes of the message route and check the correctness of
+  // transmission.
+  neo.fs.v2.session.RequestVerificationHeader verify_header = 3;
+}
+
+// SearchV2 response
+message SearchV2Response {
+  // OID with additional requested metadata.
+  message OIDWithMeta {
+    // Object ID that matches search criteria.
+    neo.fs.v2.refs.ObjectID id = 1;
+    // List of attribute data from the respective object, fields
+    // strictly follow requested ones.
+    repeated string attributes = 2;
+  }
+
+  // Main result structure.
+  message Body {
+    // List of object IDs with additional requested attributes.
+    repeated OIDWithMeta result = 1;
+
+    // Cursor that can be used for subsequent requests, an opaque string
```

Comment:
That's for clients, true, just a token. Server behavior can be documented, but it can change from version to version at the same time, so I'm not sure about that. But interoperability question is a valid one.

Reply:
> it can change from version to version

lucky we have `version` field in the request